tools/hangar/__snapshots__/test_corpus/sdk_tests/bucket/events.w_compile_tf-aws.md

```diff
@@ -1,5 +1,51 @@
 # [events.w](../../../../../../examples/tests/sdk_tests/bucket/events.w) | compile | tf-aws
 
+## CheckHitCountOptions.Struct.js
+```js
+module.exports = function(validator_req, fromInline) {
+  class CheckHitCountOptions {
+    static _schema = {
+      id: "/CheckHitCountOptions",
+      type: "object",
+      properties: {
+        key: { type: "string" },
+        type: { type: "string" },
+        source: { "$ref": "/Source" },
+        count: { type: "number" },
+      },
+      required: [
+        "key",
+        "type",
+        "source",
+        "count",
+      ]
+    }
+    static _getDependencies() {
+      return {
+        "/Source": require("./Source.Struct.js")()._schema,
+      }
+    }
+    static _getValidator() {
+      const v = new validator_req.Validator();
+      const addDependencies = ((deps) => { for (const [key, value] of Object.entries(deps)) {
+        v.addSchema(value, key);
+      }});
+      addDependencies(this._getDependencies());
+      return v;
+    }
+    static validate(obj) {
+      const result = this._getValidator().validate(obj, this._schema);
+      return result.errors;
+    }
+    static _toInflightType() {
+      return fromInline(`require("./CheckHitCountOptions.Struct.js")(require("jsonschema"))`);
```

Comment:
Any chance this can be replaced with `__filename`? Something like:
```suggestion
      return fromInline(`require("./"+__filename)(require("jsonschema"))`);
```
It'll be great if we can move all this boilerplate somewhere instead of duplicating it in every struct file, not critical though.